pkgs/dart_mcp/example/workflow_client.dart

```diff
@@ -405,39 +383,36 @@ final class WorkflowClient extends MCPClient with RootsSupport {
   }
 }
 
-final class WorkflowChatBotServer extends MCPServer with ToolsSupport {
-  final WorkflowClient client;
-
-  WorkflowChatBotServer(this.client, {required super.channel})
-    : super.fromStreamChannel(
-        implementation: ServerImplementation(
-          name: 'Gemini Chat Bot',
-          version: '0.1.0',
-        ),
-        instructions:
-            'This server handles the specific tool interactions built '
-            'into the gemini chat bot.',
-      ) {
-    registerTool(exitTool, (_) async {
-      print('goodbye!');
-      exit(0);
-    });
-  }
-
-  static final exitTool = Tool(name: 'exit', inputSchema: Schema.object());
-}
-
 final systemInstructions = gemini.Content.system('''
 You are a developer assistant for Dart and Flutter apps. You are an expert
 software developer.
 
-You can help developers by connecting into the live state of their apps, helping
+You can help developers with writing code by generating Dart and Flutter code or
+making changes to their existing app. You can also help developers with
+debugging their code by connecting into the live state of their apps, helping
 them with all aspects of the software development lifecycle.
 
-If a user asks about an error in the app, you should have several tools
-available to you to aid in debugging, so make sure to use those.
+If a user asks about an error or a widget in the app, you should have several
+tools available to you to aid in debugging, so make sure to use those.
+
+If a user asks for code that requires adding or removing a dependency, you have
+several tools available to your for managing pub dependencies.
+
+When a user asks you to complete a task, you should first make a plan, which may
+involve multiple steps and the use of tools available to you. Report this plan
+back to the user before proceeding.
+
+Generally, if you are asked to make code changes, you should follow this high
+level process:
+
+1) Write the code and apply the changes to the codebase
+2) Check for static analysis errors and warnings
+3) Fix all errors and warnings that you can
+4) Hot reload the changes to the running app
+5) Check for runtime errors
+6) Fix any runtime errors that you can
```

Comment:
Should it ask the user at this point to try out the app, and then it can listen for any additional runtime errors, or the user can give other feedback before proceeding?

Reply:
Once it is done it does ask if there is anything else the user would like. Often times the steps it generates checks for runtime errors a second time after applying changes.